ROSNET/DataModel/ROSbag.cs

```diff
@@ -1,22 +1,20 @@
-﻿using System;
-using System.IO;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 
 namespace ROSNET.DataModel
 {
+    /// <summary>
+    /// Represents a ROSbag
+    /// </summary>
     public class ROSbag
     {
-        private BinaryReader Reader { get; set; }
-        public Dictionary<int, Connection> Connections { get; private set; }
+        public Dictionary<int, Connection> Connections { get; private set; } //List of connection records in rosbag
 
+        /// <summary>
+        /// Creates an empty rosbag
+        /// </summary>
         public ROSbag()
```

Comment:
Dette er en klasse jeg ville sett over en gang til. Husk at det er dette som er det konsumenten til denne C# pakken kommer til å servert når den kaller ReadRosBag. Så pass på at alle properties og methoder som er public gir mening for sluttbrukeren. Det samme gjelder alle models som brukeren kan nå fra denne klassen. Gir det for eksempel mening for brukeren at det er en AddConnection methode? Methoder som trenges i andre klasser, men som ikke gir noe merverdi for sluttbrukeren kan for eksempel redesignes eller gjøres internal istedenfor public

Reply:
Ville kopiert `Bag`-klassen fra rosbag-Python, er ikke noe vits i at den skal bære noe særlig annerledes fra den, https://github.com/ros/ros_comm/blob/noetic-devel/tools/rosbag/src/rosbag/bag.py
Merk at med måten den håndterer filen den leser fra, så skal `Bag`-klassen i C# implementere IDisposable https://docs.microsoft.com/en-us/dotnet/api/system.idisposable?view=net-6.0, slik at filen disponeres skikkelig (lukkes) etter bruk

Reply:
Samme med tester, bare å copy-paste fra https://github.com/ros/ros_comm/tree/noetic-devel/test/test_rosbag/test

Reply:
Slik jeg har tenkt så kan RosBag objektet i fremtiden brukes til skriving også, ikke bare lesing. Dermed tenker jeg ihvertfall at AddConnection gir mening å ha tilgang til, ettersom du da kan bygge opp et RosBag objekt i fremtiden, for så å kalle en "Save(path)" metode eller noe slikt på objektet

Reply:
Det gir ikke mening å ha `AddConnection`, når hverket C++, eller Python-API-et har det.

Reply:
Se: https://github.com/ros/ros_comm/blob/noetic-devel/tools/rosbag_storage/include/rosbag/bag.h#L17. Vil gjette på at det er ugyldig å ha en `Connection` i bag-en uten å også ha en melding

Reply:
Ja ok ser hva du mener. Kan erstattes med "Write", som legger til message i connection, og automatisk oppretter connection om den ikke finnes, slik at det matcher med Python implementasjon

Reply:
Men Python sin Bag implementasjon virker til å ha en rekke funksjoner som ikke er veldig nyttig for en sluttbruker, så tror ikke at å kopiere over alt derfra er en god ide

Reply:
nvm, ser nå at det var masse private metoder